### DISCUSSION TITLE
components.xml locked from eclipse, cannot mvn clean

### DISCUSSION THREAD

**Discussion by Bananeweizen** [eclipse-tycho/tycho#1907] [Q&A]
I've meanwhile multiple times run into variants of this:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-clean-plugin:3.2.0:clean (default-clean) on project tycho-baseline-plugin: Failed to clean project: Failed to delete C:\dev\tycho\git\tycho\tycho-baseline-plugin\target\components.xml -> [Help 1]
```
when invoking `mvn clean verify` from the command line. The components.xml is locked by eclipse. I thought that components.xml is created by one of the Maven core plugins by reading the annotations from the Java source of the to be compiled Tycho module, right? If so, why would that file be locked?

Anyone else with similar experience?

**Comment by laeubi** [comment]
it could be that m2e connector (or the plugin itself) not closing the file correctly so one needs to find out if the plugin uses build-api or has a connector and then investigste the code that writes the component.xml if there might be an issue with not  closing a stream.

**Reply by Bananeweizen** [reply]
Thanks, that give me some more things to look at. Still, I'm really confused. I understood this is somehow related to Plexus Containers, since https://codehaus-plexus.github.io/plexus-containers/plexus-container-default/plexus-components.html fits quite well. However, Plexus Containers as Codehaus project has been moved after the Codehaus breakdown, and it looks like this ended in Eclipse Sisu. That in turn is used somehow in Tycho, so I'm currently not sure whether I would need to look at something in the Tycho repo, or whether that is contained somehow in one of the involved Maven plugins as binary. If I knew a class implementing the serialization of the components XML, I would probably be able to dig deeper, but all my poking of above mentioned things didn't lead any further. Maybe tomorrow, after a good night of sleep.

**Reply by laeubi** [reply]
You can open the Maven Lifecycle Mapping view:
![grafik](https://user-images.githubusercontent.com/1331477/209423895-53fc1f5e-2eb0-4299-bc91-bfa254a62088.png)

When you click an item you can see if and how the item is mapped:
![grafik](https://user-images.githubusercontent.com/1331477/209423955-72177afd-ab02-4ce7-a5b5-9f2f41d1fda0.png)

in this case it is mapped by the pom, so I would assume you need to investigate the mojo execution directly, when you look what version tycho currently uses there might be a good chance to find out what source repository is currently responsible for the code :-)
